@GaetanLepage this is the first package that makes use of `uv-dynamic-versioning` as part of the build process, so I assume this issue has remained undetected so far.

I tried to build this package using GitHub as a source, but it seems that `uv-dynamic-versioning` is not compatible with `hatchling`, due to not specifying its own version explicitly?

<details>
<summary>Here is the traceback</summary>

```python
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Traceback (most recent call last):
  File "/nix/store/gxp46ny33gykgxsbkyas7b4d64rd27sy-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
    main()
  File "/nix/store/gxp46ny33gykgxsbkyas7b4d64rd27sy-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in main
    json_out["return_val"] = hook(**hook_input["kwargs"])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/gxp46ny33gykgxsbkyas7b4d64rd27sy-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
    return _build_backend().build_wheel(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/build.py", line 58, in build_wheel
    return os.path.basename(next(builder.build(directory=wheel_directory, versions=['standard'])))
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/builders/plugin/interface.py", line 90, in build
    self.metadata.validate_fields()
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 265, in validate_fields
    _ = self.version
        ^^^^^^^^^^^^
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 149, in version
    self._version = self._get_version()
                    ^^^^^^^^^^^^^^^^^^^
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 248, in _get_version
    version = self.hatch.version.cached
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 1456, in cached
    raise type(e)(message) from None
RuntimeError: Error getting the version from source `uv-dynamic-versioning`: Unable to detect version control system. Checked: Git. Not installed: Mercurial, Darcs, Subversion, Bazaar, Fossil, Pijul.

ERROR Backend subprocess exited when trying to invoke build_wheel
error: builder for '/nix/store/wzhchk8rk8fkgb22h7wvyykbdw80dacn-uv-sort-0.5.0.drv' failed with exit code 1;
       last 25 log lines:
       >   File "/nix/store/gxp46ny33gykgxsbkyas7b4d64rd27sy-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in main
       >     json_out["return_val"] = hook(**hook_input["kwargs"])
       >                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/gxp46ny33gykgxsbkyas7b4d64rd27sy-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
       >     return _build_backend().build_wheel(
       >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/build.py", line 58, in build_wheel
       >     return os.path.basename(next(builder.build(directory=wheel_directory, versions=['standard'])))
       >                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/builders/plugin/interface.py", line 90, in build
       >     self.metadata.validate_fields()
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 265, in validate_fields
       >     _ = self.version
       >         ^^^^^^^^^^^^
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 149, in version
       >     self._version = self._get_version()
       >                     ^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 248, in _get_version
       >     version = self.hatch.version.cached
       >               ^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/nix/store/pcl9w887942hvk75rwgrbz5vnyqvvwqp-python3.12-hatchling-1.27.0/lib/python3.12/site-packages/hatchling/metadata/core.py", line 1456, in cached
       >     raise type(e)(message) from None
       > RuntimeError: Error getting the version from source `uv-dynamic-versioning`: Unable to detect version control system. Checked: Git. Not installed: Mercurial, Darcs, Subversion, Bazaar, Fossil, Pijul.
       >
       > ERROR Backend subprocess exited when trying to invoke build_wheel
       For full logs, run 'nix log /nix/store/wzhchk8rk8fkgb22h7wvyykbdw80dacn-uv-sort-0.5.0.drv'.
```
</details>

--- a/pkgs/by-name/uv/uv-sort/package.nix
+++ b/pkgs/by-name/uv/uv-sort/package.nix
@@ -1,0 +1,42 @@
+{
+  lib,
+  fetchPypi,
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "uv-sort";
+  version = "0.5.0";
+  pyproject = true;
+
+  # Build from GitHub does not work. Use fetchPypi instead of fetchFromGitHub.
+  # See https://github.com/NixOS/nixpkgs/pull/388382#issuecomment-2708857805
+  src = fetchPypi {
+    pname = "uv_sort";
+    inherit version;
+    hash = "sha256-qCShDuKBFS4omcsntZ1wzRtAKTbp8CfTw0IIAgxBvcE=";
+  };
+
+  build-system = with python3Packages; [
+    hatchling
+    uv-dynamic-versioning
+  ];
+
+  dependencies = with python3Packages; [
+    packaging
+    tomlkit
+    typer
+  ];
+
+  nativeCheckInputs = [
+    python3Packages.pytestCheckHook
+  ];
+
+  meta = {
+    description = "Sort uv's dependencies alphabetically";
+    homepage = "https://github.com/ninoseki/uv-sort";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ aaronjheng ];
+    mainProgram = "uv-sort";
+  };
+}

